File: TeAchei/src/main/java/com/teachei/api/application/ports/in/GerenciarPerfilUseCase.java
package com.teachei.api.application.ports.in;

import com.teachei.api.domain.model.Perfil;

import java.util.UUID;

/**
 * Inbound port for profile management use case.
 */
public interface GerenciarPerfilUseCase {

    /**
     * Gets the profile of a user.
     *
     * @param usuarioId the user ID
     * @return the user's profile
     */
    Perfil buscarPorUsuario(UUID usuarioId);

    /**
     * Updates a user's profile.
     *
     * @param usuarioId the user ID making the request
     * @param command the update command
     * @return the updated profile
     */
    Perfil atualizar(UUID usuarioId, AtualizarPerfilCommand command);

    /**
     * Command for profile update.
     */
    record AtualizarPerfilCommand(
        String nome,
        String bio,
        String whatsapp,
        String instagram,
        String facebook,
        String cidade,
        String estado,
        String role
    ) {}
}

File: teachei-web/ap/perfil/application/ports/in/GerenciarPerfilUseCase.ts
import type { Perfil, AtualizarPerfilInput } from "@/ap/perfil/domain/model/Perfil";

export interface GerenciarPerfilUseCase {
  buscar(usuarioId: string): Promise<Perfil>;
  buscarPorId(id: string): Promise<Perfil>;
  atualizar(usuarioId: string, data: AtualizarPerfilInput): Promise<Perfil>;
  criarOuBuscar(usuarioId: string, nome: string): Promise<Perfil>;
}
